src/main/java/randoop/generation/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,254 @@
+package randoop.generation;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.checkerframework.checker.signedness.qual.Signed;
+import randoop.main.GenInputsAbstract;
+import randoop.sequence.Sequence;
+import randoop.util.Log;
+
+/**
+ * This class stores constant mining information. T is the scope of the constant mining, which can
+ * be ClassOrInterfaceType, Package, or Object, which corresponds to users' input about literal
+ * level as CLASS, PACKAGE, or ALL. The storage stores the frequency of the sequence, the number of
+ * classes that contain the sequence, and the total number of classes in the current scope.
+ *
+ * @param <T> the scope of the constant mining
+ */
+public class ConstantMiningStatistics<T extends @Signed Object> {
+
+  /**
+   * A map from a specific scope to its frequency information, which stands for the number of times
```

Comment:
For each of the maps (which will be field in ScopeStatistics), please document when the value can be null.